File: src/lib/api/showroom.ts
/**
 * Showroom API client - Backend ile iletişim
 */

const API_BASE_URL = process.env.NEXT_PUBLIC_API_URL || 'https://backend-mlh.heni.com.tr';

export interface Product {
  id: string;
  name: string;
  slug: string;
  sku?: string;
  short_description?: string;
  description?: string;
  category?: string;
  brand?: string;
  is_active: boolean;
  moq?: number;
  image_url?: string;
  technical_specs?: Record<string, any>;
  features?: string[];
  media?: ProductMedia[];
  documents?: ProductDocument[];
  export_countries?: ExportCountry[];
  created_at?: string;
  updated_at?: string;
}

export interface ProductMedia {
  id: string;
  media_url: string;
  media_type: 'image' | 'video';
  media_category?: 'product' | 'loading' | 'certificate' | 'technical' | 'lifestyle';
  display_order: number;
}

export interface ProductDocument {
  id: string;
  document_type: 'msds' | 'coa' | 'analysis_report' | 'certificate' | 'spec_sheet' | 'other';
  title: string;
  file_url: string;
  file_name: string;
  file_size?: number;
  language: string;
}

export interface ExportCountry {
  id: string;
  country_code: string;
  country_name: string;
  flag_url?: string;
  compliance_notes?: string;
  hs_code?: string;
  display_order: number;
}

export interface CartItem {
  id: string;
  cart_id: string;
  product_id: string;
  variant_id?: string;
  quantity: number;
  product_name: string;
  product_sku?: string;
  product_image_url?: string;
  variant_name?: string;
  unit_price?: number;
  created_at: string;
  updated_at: string;
}

export interface Cart {
  id: string;
  session_id: string;
  items: CartItem[];
  created_at: string;
  updated_at: string;
}

export interface QuoteRequest {
  cart_id?: string;
  items: {
    product_id: string;
    variant_id?: string;
    quantity: number;
  }[];
  company_name: string;
  contact_person: string;
  email: string;
  phone: string;
  country: string;
  notes?: string;
}

export interface QuoteResponse {
  id: string;
  quote_number: string;
  status: string;
  created_at: string;
  message: string;
}

// API fonksiyonları
export const showroomAPI = {
  // Ürünleri listele
  async getProducts(params?: {
    category?: string;
    search?: string;
    limit?: number;
    offset?: number;
  }) {
    const queryParams = new URLSearchParams();
    if (params?.category) queryParams.append('category', params.category);
    if (params?.search) queryParams.append('search', params.search);
    if (params?.limit) queryParams.append('limit', params.limit.toString());
    if (params?.offset) queryParams.append('offset', params.offset.toString());
    
    const response = await fetch(`${API_BASE_URL}/api/showroom/products?${queryParams}`);
    if (!response.ok) throw new Error('Ürünler yüklenemedi');
    return response.json();
  },
  
  // Ürün detayını getir
  async getProduct(slug: string): Promise<Product> {
    const response = await fetch(`${API_BASE_URL}/api/showroom/products/${slug}`);
    if (!response.ok) throw new Error('Ürün bulunamadı');
    return response.json();
  },
  
  // Sepeti getir
  async getCart(sessionId: string): Promise<Cart> {
    const response = await fetch(`${API_BASE_URL}/api/showroom/cart/${sessionId}`);
    if (!response.ok) throw new Error('Sepet yüklenemedi');
    return response.json();
  },
  
  // Sepete ürün ekle
  async addToCart(sessionId: string, item: {
    product_id: string;
    variant_id?: string;
    quantity: number;
  }) {
    const response = await fetch(`${API_BASE_URL}/api/showroom/cart/${sessionId}/items`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(item),
    });
    if (!response.ok) throw new Error('Ürün eklenemedi');
    return response.json();
  },
  
  // Sepet ürünü güncelle
  async updateCartItem(itemId: string, quantity: number) {
    const response = await fetch(`${API_BASE_URL}/api/showroom/cart/items/${itemId}`, {
      method: 'PUT',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ quantity }),
    });
    if (!response.ok) throw new Error('Ürün güncellenemedi');
    return response.json();
  },
  
  // Sepetten ürün kaldır
  async removeFromCart(itemId: string) {
    const response = await fetch(`${API_BASE_URL}/api/showroom/cart/items/${itemId}`, {
      method: 'DELETE',
    });
    if (!response.ok) throw new Error('Ürün kaldırılamadı');
    return response.json();
  },
  
  // Teklif talebi oluştur
  async createQuote(quoteData: QuoteRequest): Promise<QuoteResponse> {
    const response = await fetch(`${API_BASE_URL}/api/showroom/quotes`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(quoteData),
    });
    if (!response.ok) throw new Error('Teklif talebi gönderilemedi');
    return response.json();
  },
};

File: backend/schemas/showroom.py
"""
Showroom ve cart için Pydantic schema'ları
"""
from pydantic import BaseModel, EmailStr, Field
from typing import Optional, List
from datetime import datetime
from uuid import UUID


# Product Schemas
class ProductMedia(BaseModel):
    """Ürün medya modeli"""
    id: UUID
    media_url: str
    media_type: str  # 'image' veya 'video'
    media_category: Optional[str] = 'product'  # 'product', 'loading', 'certificate', 'technical', 'lifestyle'
    display_order: int = 0


class ProductDocument(BaseModel):
    """Ürün belge modeli"""
    id: UUID
    document_type: str  # 'msds', 'coa', 'analysis_report', 'certificate', 'spec_sheet', 'other'
    title: str
    file_url: str
    file_name: str
    file_size: Optional[int] = None
    language: str = 'en'


class ExportCountry(BaseModel):
    """İhracat ülkesi modeli"""
    id: UUID
    country_code: str
    country_name: str
    flag_url: Optional[str] = None
    compliance_notes: Optional[str] = None
    hs_code: Optional[str] = None
    display_order: int = 0


class ProductBase(BaseModel):
    """Temel ürün bilgileri"""
    id: UUID
    name: str
    slug: str
    sku: Optional[str] = None
    short_description: Optional[str] = None
    description: Optional[str] = None
    category: Optional[str] = None
    brand: Optional[str] = None
    is_active: bool = True


class ProductListItem(ProductBase):
    """Ürün listesi için kısaltılmış model"""
    image_url: Optional[str] = None
    moq: Optional[int] = None


class ProductDetail(ProductBase):
    """Detaylı ürün modeli"""
    technical_specs: Optional[dict] = None
    features: Optional[List[str]] = None
    moq: Optional[int] = None
    media: List[ProductMedia] = []
    documents: List[ProductDocument] = []
    export_countries: List[ExportCountry] = []
    created_at: datetime
    updated_at: datetime


# Cart Schemas
class CartItemBase(BaseModel):
    """Sepet item temel model"""
    product_id: UUID
    quantity: int = Field(gt=0, description="Miktar 0'dan büyük olmalı")
    variant_id: Optional[UUID] = None


class CartItemCreate(CartItemBase):
    """Sepete ürün ekleme"""
    pass


class CartItem(CartItemBase):
    """Sepet item response"""
    id: UUID
    cart_id: UUID
    product_name: str
    product_sku: Optional[str] = None
    product_image_url: Optional[str] = None
    variant_name: Optional[str] = None
    unit_price: Optional[float] = None
    created_at: datetime
    updated_at: datetime


class CartBase(BaseModel):
    """Sepet temel model"""
    session_id: str


class CartCreate(CartBase):
    """Sepet oluşturma"""
    pass


class Cart(CartBase):
    """Sepet response"""
    id: UUID
    items: List[CartItem] = []
    created_at: datetime
    updated_at: datetime
    
    @property
    def total_items(self) -> int:
        """Toplam ürün sayısı"""
        return len(self.items)
    
    @property
    def total_quantity(self) -> int:
        """Toplam miktar"""
        return sum(item.quantity for item in self.items)


# Quote Request Schemas
class QuoteRequestCreate(BaseModel):
    """Teklif talebi oluşturma"""
    cart_id: Optional[UUID] = None
    items: List[CartItemBase]  # Alternatif olarak direkt item listesi
    company_name: str = Field(..., min_length=2, max_length=200)
    contact_person: str = Field(..., min_length=2, max_length=100)
    email: EmailStr
    phone: str = Field(..., min_length=5, max_length=20)
    country: str = Field(..., min_length=2, max_length=100)
    notes: Optional[str] = Field(None, max_length=1000)


class QuoteRequestResponse(BaseModel):
    """Teklif talebi response"""
    id: UUID
    quote_number: str
    status: str
    created_at: datetime
    message: str = "Teklif talebiniz başarıyla alındı"


# Update schemas
class CartItemUpdate(BaseModel):
    """Sepet item güncelleme"""
    quantity: int = Field(gt=0, description="Miktar 0'dan büyük olmalı")
